<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
<title>OMNeT++ Simulation Library: cStackCleanupException Class Reference</title>
<link href="doxygen.css" rel="stylesheet" type="text/css">
<link href="tabs.css" rel="stylesheet" type="text/css">
</head><body>
<!-- Generated by Doxygen 1.5.5 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
</div>
<div class="contents">
<h1>cStackCleanupException Class Reference<br>
<small>
[<a class="el" href="group__Internals.html">Internal classes</a>]</small>
</h1><!-- doxytag: class="cStackCleanupException" --><!-- doxytag: inherits="cException" --><code>#include &lt;<a class="el" href="cexception_8h-source.html">cexception.h</a>&gt;</code>
<p>
<div class="dynheader">
Inheritance diagram for cStackCleanupException:</div>
<div class="dynsection">

<p><center><img src="classcStackCleanupException.png" usemap="#cStackCleanupException_map" border="0" alt=""></center>
<map name="cStackCleanupException_map">
<area href="classcException.html" alt="cException" shape="rect" coords="0,0,152,24">
</map>
</div>

<p>
<a href="classcStackCleanupException-members.html">List of all members.</a><hr><a name="_details"></a><h2>Detailed Description</h2>
Used internally when deleting an activity() simple module. 
<p>
Then, the coroutine running activity() is "asked" to throw a <a class="el" href="classcStackCleanupException.html" title="Used internally when deleting an activity() simple module.">cStackCleanupException</a> to achieve stack unwinding, a side effect of exceptions, in order to properly clean up activity()'s local variables. <table border="0" cellpadding="0" cellspacing="0">
<tr><td></td></tr>
<tr><td colspan="2"><br><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcStackCleanupException.html#7b8cb032ed9b1b547c7f654d693180e3">cStackCleanupException</a> ()</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcStackCleanupException.html#c7b9457b6cab0a8fc2e9f5601c85fdeb">cStackCleanupException</a> (const <a class="el" href="classcStackCleanupException.html">cStackCleanupException</a> &amp;e)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual <a class="el" href="classcStackCleanupException.html">cStackCleanupException</a> *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcStackCleanupException.html#6106d0f66dee0d823e1f6f4868c9ea65">dup</a> () const </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual bool&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcStackCleanupException.html#b7728cfc9eb1e9d507f9ddbc7163f580">isError</a> () const </td></tr>

</table>
<hr><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" name="7b8cb032ed9b1b547c7f654d693180e3"></a><!-- doxytag: member="cStackCleanupException::cStackCleanupException" ref="7b8cb032ed9b1b547c7f654d693180e3" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cStackCleanupException::cStackCleanupException           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Default ctor. 
<p>

</div>
</div><p>
<a class="anchor" name="c7b9457b6cab0a8fc2e9f5601c85fdeb"></a><!-- doxytag: member="cStackCleanupException::cStackCleanupException" ref="c7b9457b6cab0a8fc2e9f5601c85fdeb" args="(const cStackCleanupException &amp;e)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cStackCleanupException::cStackCleanupException           </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classcStackCleanupException.html">cStackCleanupException</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>e</em>          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
We unfortunately need to copy exception objects when handing them back from an activity(). 
<p>

</div>
</div><p>
<hr><h2>Member Function Documentation</h2>
<a class="anchor" name="6106d0f66dee0d823e1f6f4868c9ea65"></a><!-- doxytag: member="cStackCleanupException::dup" ref="6106d0f66dee0d823e1f6f4868c9ea65" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual <a class="el" href="classcStackCleanupException.html">cStackCleanupException</a>* cStackCleanupException::dup           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"> const<code> [inline, virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Virtual copy constructor. 
<p>
We unfortunately need to copy exception objects when handing them back from an activity(). 
<p>Reimplemented from <a class="el" href="classcException.html#7aae9b928c595645f2918a897eb14c2b">cException</a>.</p>

</div>
</div><p>
<a class="anchor" name="b7728cfc9eb1e9d507f9ddbc7163f580"></a><!-- doxytag: member="cStackCleanupException::isError" ref="b7728cfc9eb1e9d507f9ddbc7163f580" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual bool cStackCleanupException::isError           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"> const<code> [inline, virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
This exception type does not represent an error condition. 
<p>

<p>Reimplemented from <a class="el" href="classcException.html#e94e8963aaae31fc8609986128d38d52">cException</a>.</p>

</div>
</div><p>
<hr>The documentation for this class was generated from the following file:<ul>
<li><a class="el" href="cexception_8h-source.html">cexception.h</a></ul>
</div>
<hr size="1"><address style="text-align: right;"><small>Generated on Tue Mar 27 11:22:13 2012 for OMNeT++ Simulation Library by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 1.5.5 </small></address>
</body>
</html>
